packages/orchestration/src/workflow/local-workflow.ts

```diff
@@ -39,17 +39,17 @@ export class LocalWorkflow {
     this.workflow = globalWorkflow
     this.handlers = new Map(globalWorkflow.handlers_)
 
-    const container = createMedusaContainer()
+    let container
 
-    // Medusa container
     if (!Array.isArray(modulesLoaded) && modulesLoaded) {
-      const cradle = modulesLoaded.cradle
-      for (const key of Object.keys(cradle ?? {})) {
-        container.register(key, asValue(cradle[key]))
+      if (!("cradle" in modulesLoaded)) {
+        container = createContainerLike(modulesLoaded)
+      } else {
+        container = modulesLoaded
```

Comment:
More specifically, the util and why this was needed

Reply:
The workflow function expect a "container" which is basically an object with a resolve function and return something. In reality, we will have to change the types to be more generic and not tight to awilix specifically. As a first step here, we fix the issues where sometimes we don't have a container at hand but the cradle which is a proxy object which calls the resolve under the hood. The utils allow to wrap any object and return an object with a resolve function which will use the provided object to resolve the expected return. Does it make sense?

Reply:
Before, the workflow only accepted a container. The strategies like import and export don't have access to a container, they only have access to a cradle, which is a container proxy.

Now, if you pass in a container proxy, which has everything that the workflow needs, we will convert into a container like shape instead of recreating a new container just for the purpose of the workflow.

Reply:
Thanks for the explanation guys. This is very clear 👍